File: foliage-api/src/main/java/org/lavenderx/foliage/api/controller/HelloController.java
package org.lavenderx.foliage.api.controller;

import lombok.extern.slf4j.Slf4j;
import org.lavenderx.foliage.nettyrpc.rpcservice.HelloRpcService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping
@Slf4j
public class HelloController {

    @Autowired
    private HelloRpcService helloRpcService;

    @GetMapping(path = "/hello")
    public ResponseEntity<String> hello() {
        String result = helloRpcService.hello("World");
        log.info("Rpc call result: {}", result);
        return new ResponseEntity<>(result, HttpStatus.OK);
    }
}

File: foliage-api/src/main/kotlin/org/lavenderx/foliage/api/controller/HelloController.kt
package org.lavenderx.foliage.api.controller

import io.foliage.utils.loggerFor
import io.foliage.netty.rpc.rpcservice.HelloRpcService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping
class HelloController @Autowired constructor(val helloRpcService: HelloRpcService) {

    private val logger = loggerFor<HelloController>()

    @GetMapping("/hello")
    fun hello(): ResponseEntity<String> {
        val result = helloRpcService.hello("World")
        logger.info("Rpc call result: {}", result)
        return ResponseEntity(result, HttpStatus.OK)
    }
}